src/rpc/connectivity/cluster.cc

```diff
@@ -1028,7 +1058,7 @@ void connectivity_cluster_t::run_t::handle(
                 "server with the `--initial-password auto` option to allow joining "
                 "the password-protected cluster.");
             fail_handshake(conn, peername, reason);
-            return;
+            return join_result_t::PERMANENT_ERROR;
```

Comment:
To be conservative, we might want to keep this as a TEMPORARY_ERROR.
A user could connect to either this or the other server, and reset or set the admin password respectively, without restarting it.